recipes/full_finetune_distributed.py

```diff
@@ -357,10 +357,12 @@ def _setup_data(
         iterable datasets and streaming datasets are not supported.
         """
         world_size, rank = utils.get_world_size_and_rank()
-        ds = config.instantiate(
-            cfg_dataset,
-            tokenizer=self._tokenizer,
-        )
+
+        if isinstance(cfg_dataset.get(0), DictConfig):
```

Comment:
you can also just directly check if it's a ListConfig. If it's a single dataset then this might fail
```suggestion
        if isinstance(cfg_dataset, ListConfig):
```

I also wonder if there's a better way to handle this so we don't have to repeat this if-else check across all recipes...

Reply:
Good point, i've replaced it with `ListConfig` check.

Reply:
> I also wonder if there's a better way to handle this so we don't have to repeat this if-else check across all recipes...

Yeah, this logic can be moved to `config.instantiate` method, but I guess it will break single responsibility principle.